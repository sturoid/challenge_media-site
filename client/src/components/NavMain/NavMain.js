import React from 'react';
import { NavLink } from 'react-router-dom';
import { wrapper1024 } from '../../lib/wrappers.styles';
import { header, h1, navItems, navLink } from './NavMain.styles';

const NavMain = () => (
  <header className={header}>
    <div className={wrapper1024}>
      <div className={navItems}>
        <h1 className={h1}>Behance User Search</h1>
        <NavLink className={navLink} to="/" exact>
          Search
        </NavLink>
      </div>
    </div>
  </header>
);

export default NavMain;
