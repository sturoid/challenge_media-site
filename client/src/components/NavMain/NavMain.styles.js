import { css } from 'emotion';
import { greyDark, white } from '../../lib/colors.styles';

export const header = css`
  background-color: ${greyDark};
  color: ${white};
  padding: 10px 0;
`;

export const h1 = css`
  font-weight: normal;
  font-size: 1.3rem;
  margin: 0;
`;

export const navItems = css`
  display: flex;
  align-items: center;
`;

export const navLink = css`
  color: white;
  text-decoration: none;
  padding: 0 20px;
`;
