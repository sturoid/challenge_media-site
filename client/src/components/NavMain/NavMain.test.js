import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavMain from './NavMain';

describe('<UserNav />', () => {
  it('renders correctly and equals snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <NavMain />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
