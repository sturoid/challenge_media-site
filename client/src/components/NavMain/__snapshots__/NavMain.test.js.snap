// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserNav /> renders correctly and equals snapshot 1`] = `
<header
  className="css-vyx1fs"
>
  <div
    className="css-vzsrkt"
  >
    <div
      className="css-1sg2lsz"
    >
      <h1
        className="css-xbu0kd"
      >
        Behance User Search
      </h1>
      <a
        aria-current="page"
        className="css-1tl4qs6 active"
        href="#/"
        onClick={[Function]}
        style={Object {}}
      >
        Search
      </a>
    </div>
  </div>
</header>
`;
